### DISCUSSION TITLE
CANT DELETE USERS ON SUPABASE & NextJS

### DISCUSSION THREAD

**Discussion by fizzyceo** [supabase/supabase#14032] [Questions]
1st issue: I'm trying to access the email_confirmed_At column in the auth.user table but can't because it is locked by default 
![image](https://user-images.githubusercontent.com/84289623/235299490-9807f2b0-7472-47bc-a5e2-5b7cb06a0ac7.png)
so I want to extract only that column and use it in my public.users table to get a reference of who is verified and who isnt

2nd issue: I can't delete users properly with the new version of SUPABASE, as the method supabase.auth.api.deleteUser no longer exists, so even if i delete a user in my public. user table it will still exist in supabase and he can log in again.
I need a solution ASAP please :)
thank you.

**Comment by GaryAustin1** [comment]
1). Not sure what locked by default means. You can read auth.users with a security definer Postgres function or from rpc or a trigger.  

2). https://supabase.com/docs/reference/javascript/auth-admin-deleteuser   Delete user.

**Reply by fizzyceo** [reply]
thank you for your response for the first issue i fixed it by running a query that grants access to all users  aka "public", then i was able to access the auth.users table from the API I also added a new supabase admin that uses service_role key 
`import { createClient } from '@supabase/supabase-js';

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL ;
const supabaseKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY ;
const supabaseSecret = process.env.NEXT_PUBLIC_SUPABASE_SERVICE_ROLE_KEY ;


const supabase = createClient(supabaseUrl, supabaseKey);

console.log(supabaseSecret,supabaseKey);
export const supabaseAdmin = createClient(supabaseUrl, supabaseSecret);
export default supabase;`
so that i could use it in my handler
`export default async function handler(req, res) {
    const {userid} = JSON.parse(req.body);
    const { data, error } = await supabase
    .from('user')
    .delete()
    .eq('user_id', userid)
  if (error) {
        
    return res.status(400).json({ error:"problem with the public table"})
  }
  console.log(supabaseSecret);
  const deleted_from_auth = await supabaseAdmin.auth.admin.deleteUser(userid)
  if(deleted_from_auth.error) {
    return res.status(400).json({ error:"problem with the auth table"})
  }
  console.log(deleted_from_auth.data);
    
    return res.status(200).json({ deletedUser: deleted_from_auth.data })
}
`

**Reply by GaryAustin1** [reply]
If you modified auth.users to grant public access that is a very bad idea. First Supabase can rebuild that table at any time.  Second you imply you can access from the API which would also be bad.  But not really clear exactly what you did.
